### PR TITLE
[ENH] update `TinyTimeMixerForecaster` with new model support, pretraining, and compatibility improvements

### LINKED ISSUE #7829: [BUG] Train/Test Split in GlobalForecaster (TinyTimesMixer) lead to empty  train/test sets.
**Describe the bug**

Due to the splitting it happens in the tests that the train or test sets are empty.  See: https://github.com/sktime/sktime/actions/runs/13328428251/job/37227372162?pr=7759

**To Reproduce**

```python

from build.lib.build.lib.sktime.forecasting.ttm import TinyTimeMixerForecaster
from sktime.utils.estimator_checks import check_estimator

check_estimator(TinyTimeMixerForecaster, tests_to_run=["test_global_predict_quantiles"], raise_exceptions=True)```

**Expected behavior**
Should not fail. And in case it needs to fail the error message has to be better.

**Additional context**
<!--
Add any other context about the problem here.
-->

**Versions**
<details>

<!--
Please run the following code snippet and paste the output here:

from sktime import show_versions; show_versions()
-->

</details>

<!-- Thanks for contributing! -->

### LINKED ISSUE #8770: [ENH] Update get_model() to select the best model form TTM family
**Is your feature request related to a problem? Please describe.**
TTM (R2.2) has 22 different model for different context-lengths and forecast-lengths. We have a module named  get_model() to select the best model given context-length, forecast-length and other params. We would like to have the get_model() functionality integrated.

**Describe the solution you'd like**
We would like to have the get_model() functionality integrated in the sktime. get_model() takes a few arguments including the context-length and forecast-length. Currently, this has been implemented in the [IBM-Granite-TSFM public repo](https://github.com/ibm-granite/granite-tsfm/blob/main/tsfm_public/toolkit/get_model.py). 

get_model() need to be called in the _fit() call while initializing the TTM model. Further, get_model() has some additional dependency on the TTM pre-processor, and some utilities located at [tsfm_public/toolkit](https://github.com/ibm-granite/granite-tsfm/tree/main/tsfm_public/toolkit). Theses dependencies should also be copied to [/sktime/sktime/libs/granite_ttm/](https://github.com/kr-saurav-010/sktime/tree/ttm-update-get-model/sktime/libs/granite_ttm). 

I have started to work on this, will raise a working pull request soon !

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #9737: [ENH] Add TTM-R3 support to TinyTimeMixerForecaster
IBM Research recently released TTM-R3 (`ibm-research/ttm-r3` on HuggingFace).
The current `TinyTimeMixerForecaster` supports R1 (`ibm/TTM`) but not R3.
This issue proposes adding R3 support.

I compared R3's `config.json` with the current `TinyTimeMixerConfig` in sktime and found new fields that are not yet supported, including:

Trend–residual decomposition
Probabilistic forecasting heads
New loss configurations
FFT-based embeddings

The `model_type` in R3's config is still `"tinytimemixer"`, so the same loading interface applies.

## Questions

1. Should `configuration_tinytimemixture` in sktime be updated to support R3's new fields, right?
2. R3 license is `cc-by-nc-sa-4.0` (non-commercial). Does this affect how we interface it compared to current version?
3. Should R3 reuse `TinyTimeMixerForecaster` with `model_path="ibm-research/ttm-r3"`, or get a separate class?

cc @geetu040 @fkiraly @RecreationalMath

### PR DESCRIPTION
**Pull request by geetu040**
This PR updates and extends the `TinyTimeMixerForecaster` implementation with support for newer TinyTimeMixer model families, pretraining workflows, unequal-length data, and several usability, compatibility, and documentation improvements.

Towards #10153, #10152
Closes #9737, #8770, #7829
And other enhancements not tracked in separate issues.

The changes are independent of #9664 (migration to the granite-tsfm PyPI package) and can be reviewed in parallel.

**Changes**

* Add support for `ttm-r3` models.
* Add support for `ttm-research-r2` models.
* Add support for additional `ttm-r2` variants that were previously unsupported (e.g. `52-16-ft-r2.1`).
* Implement automatic model selection when `revision=None`.
* Add `frequency_token` support required by newer R2/R3 model families.
* Enable estimator pretraining functionality.
* Enable estimator caching functionality.
* Add support for unequal-length series on pretraining.
* Enable fitting without requiring `fh` during `fit`.
* Fix compatibility issues when using `use_source_package=True`.
* Update the vendored `sktime.libs.granite_ttm` fork to match newer upstream changes.
* Update forked code for compatibility with newer Transformers versions, including v5-related changes.
* Verify estimator functionality across CI environments and supported execution configurations.
* Add a `device` parameter for explicit device selection.
* Add a `verbose` parameter for improved debugging and monitoring.
* Update test parameters to reduce test execution time.
* Add support for unequal-length collections by internally handling padding/truncation where required.
* Expand references to TinyTimeMixer papers, model collections, and related resources.
* Add additional usage examples covering different model families and configurations.

### CODE REVIEW

**Comment by geetu040** [comment]
All the tests are passing for TTM in the CI from recent commit: [runs/27404719539/](https://github.com/sktime/sktime/actions/runs/27404719539/).
So nothing is fundamentally broken in the interface.

**Review by fkiraly — APPROVED** [review]
